### ISSUE TITLE
Create TfDatasetFactory for Muilti-Output Keras in _input_fn

### ISSUE THREAD

**Issue by wlee192** [tensorflow/tfx#3207]
## URL(s) with the issue:

https://github.com/tensorflow/tfx/blob/master/tfx/examples/chicago_taxi_pipeline/taxi_utils_native_keras.py
https://www.tensorflow.org/tfx/tutorials/tfx/components_keras#trainer

### Clear description:

Hi, i've been following the tutorial for chicago taxi example for tfx, using keras models. 
I am now trying to apply it to my own dataset. 

The difference here, being that instead of a single output, the keras model i have has 2 outputs, built based on functional API, as the task at hand requires the model to output 2 different set of classification for a given input.

I've tried to look around but could not find any example yet. 

The current taxi scenario (https://github.com/tensorflow/tfx/blob/master/tfx/examples/chicago_taxi_pipeline/taxi_utils_native_keras.py) outputs tensor with a single label in the `_input_fn`, and i don't think the  `dataset_options.TensorFlowDatasetOptions()` function allows user to pass in 2 labels (given that i have 2 outputs in the model).

I have also posted this query on stackoverflow with more details of what i've tried (https://stackoverflow.com/questions/65879049/multioutput-classification-with-tensorflow-extended-tfx) but have not heard back yet.

Are you able to point me to an example/lead where i can build/modify the `_input_fn` to output tensor of tuple `(dict[features], dict[labels])` ; where the labels consist of 2 outputs, and is compatible with the tfx pipeline? If you can shed more light on this through a code example that would be even better. Thanks!

### Parameters defined

Yes

### Returns defined

Yes

### Raises listed and defined

Are the errors defined? For example,

refer to the stackoverflow link for more detail

### Request visuals, if applicable

refer to the stackoverflow link for more detail

**Comment by brills** [comment]
If I understand the problem correctly (i.e. there are two features in your tf.Examples that should be labels used by the model, and you want to build a Dataset that yields `(rest_of_features, label_dict)`), you can simply leave  `TensorFlowDatasetOptions.label_key` to be `None`. This way, elements of the Dataset you will get are dicts (instead of tuples), then you can do something like:

```
def prepare_label(feature_dict):
  label_dict = {}
  label_dict['label1'] = feature_dict.pop("label1")
  label_dict['label2'] = feature_dict.pop("label2")
  return feature_dict, label_dict

dataset = data_accessor.tf_dataset_factory(...)
dataset = dataset.map(prepare_label)

```

**Comment by wlee192** [comment]
Hi @brills ,

Thanks for the suggestion. 
That's another neat way to work with it. 

I've tried the above and it's working for me. 
I'll close this issue out based on the suggestion provided. 

Thanks
